[STM32F0xx] Can development for STM32F0xx family

BartSX opened the pull request:
CAN development for STM32F0xx family

0xc0170 commented on libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F0/can_api.c:
```diff
@@ -0,0 +1,458 @@
+    }  
+}
+
+void CEC_CAN_IRQHandler(void)
```
is this weak IRQ symbol? You should use your won irq handler, register it via ``NVIC_SetVector``

adustm commented:
Hello @BartSX ,

In libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F0/TARGET_NUCLEO_F091RC/PeripheralPins.c and libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F0/TARGET_NUCLEO_F072RB/PeripheralPins.c
Could you delete the lines that are not part of the packaging of the platform ? (the packaging is fixed to LFQP64 on the NUCLEO64 board). We don't mention other packaging, even in commented lines.
`// Available only in UFBGA100 and LQFP100 package`

adustm commented:
Hello @BartSX 
In libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F0/TARGET_NUCLEO_F091RC/PeripheralPins.c and libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F0/TARGET_NUCLEO_F072RB/PeripheralPins.c
`//  {PB_8 , CAN_1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_NOPULL, GPIO_AF4_CAN)}` 
`//  {PB_9 ,  CAN_1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_NOPULL, GPIO_AF4_CAN)},`
Could you clarify why you have commented PB_8 and PB_9 pins ? Looking at the schematics, I could not find the reason.

BartSX commented:
Hi @adustm and @0xc0170 

Sure, I will delete those lines.
When it comes to PB_8 and PB_9, you are right, there's no reason for them to be commented, it's my typo. I will update the change later.

@0xc0170,
`NVIC_SetVector` is being used in `can_irq_set()` and the name can be confusing but that does not affect the overall CAN behavior at all.

0xc0170 commented:
Just rename from ``CEC_CAN_IRQHandler`` to something like ``can_irq_handler``. Otherwise it brings the confusion as you noted.

0xc0170 commented:
Bump, what do you think about my proposal above?

BartSX commented:
Yes, I do agree. I will make the change but I'm waiting with the update because I want to implement [#1735](https://github.com/mbedmicro/mbed/pull/1735) into F0 family as well.

BartSX commented:
@0xc0170,
I've updated the F0xx family and added one more Nucleo (F042K6). Please review, because I'd like to push other families and that require merging *tests.py* file.
Thanks

adustm commented:
Hello @BartSX ,
Is there a problem while testing PB_8 and PB_9 ? I can see they are still commented even in the new F042.
If there is no problem you need to uncomment them in CAN_RD and CAN_TD

Kind regards
Armelle

BartSX commented:
Hi @adustm,

I've updated the pull request, please review.
For F042K6 only CAN_RX PB_8 pin has to be commented because it's available on that package but not on Nucleo itself. PB_9 (CAN_TX) is not available on that board so everything should be correct.

BR,
Bartosz

adustm commented:
Hello, 
LGTM
Kind regards
Armelle

sg- commented:
👍

BartSX commented:
Hi all,
I think that we have a small problem. I've noticed that there was a directory change on 14th of May in d9734e5a but CAN F0 is merged and now it creates an unneeded directory. I will crate a separate pull request which will replace all the F0xx hal files as needed for CAN.

BR,
Bartosz

0xc0170 commented:
Yes, thanks for pinpointing, please send PR or I'll fix it, we are rebasing the current pull requests ready for integration
